contributors/design-proposals/selector-generation-for-all.md

```diff
@@ -0,0 +1,71 @@
+Selector Generation for Deployments, ReplicaSet, DaemonSet, StatefulSet and ReplicationController
+=============
+
+# Goals
+Make selector easy to use and less error-prone for Deployments, ReplicaSet, DaemonSet, StatefulSet and ReplicationController. Make `kubectl apply` work well with selector.
```

Comment:
Can we remove all wording in this doc that's about correctness? `controllerRef` should take care of the existing problems. I think there is something that needs to be done to fix `kubectl apply`, but this is a really invasive change and we should think about whether anything else can be done.